.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,231 @@
+name: "pull-request"
+
+on:
+  push:
+    branches: [master, develop]
+  pull_request:
+    branches: [master, develop]
+
+env:
+  APP_CONFIG_ECM_HOST: ${{ secrets.PIPELINE_ENV_URL }}
+  ADMIN_EMAIL: ${{ secrets.PIPELINE_ADMIN_USERNAME }}
+  ADMIN_PASSWORD: ${{ secrets.PIPELINE_ADMIN_PASSWORD }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
```

Comment:
avoid injecting sensible credentials as global envs, use the aws-actions/configure-aws-credentials action only in the jobs where you really need them.

aws region is not really a secret and should be a standard variable

can you give me a summary of what aws resources this build requires to access? we could switch to more fine grained secrets, see the ones already available in https://alfresco.atlassian.net/wiki/spaces/TECH/pages/1002439885/GitHub+Organization+secrets+for+Alfresco

the credentials you are currently using are deprecated and we'll had to stop using them anyway in the future

Reply:
For now only those 2 that are used to configure aws